js/ui/test-button.tsx

```diff
@@ -0,0 +1,208 @@
+    }
+    if(model.icon) {
+        classNames.push('dx-button-has-icon');
+    }
+
+    if (model._hovered) {
+        classNames.push('dx-state-hover');
+    }
+
+    if (model.pressed || model._active) {
+        classNames.push('dx-state-active');
+    }
+
+    if (!model.visible) {
+        classNames.push('dx-state-invisible');
+    }
+    return classNames.concat(model.classNames).join(" ");
+}
+
+const getElementAttribute = (name: string, value: string) => {
+    const validName = (name === 'role' || name === 'id') ? name : `aria-${name}`;
+    const validValue = isDefined(value) ? value.toString() : null;
+
+    const result: any = {};
+    result[validName] = validValue;
+    return result;
+};
+
+const getAttributes = (attrs: any) => {
+    let elementAttributes = {};
+
+    for (let attrName in attrs) {
+        elementAttributes = { ...elementAttributes, ...getElementAttribute(attrName, attrs[attrName]) };
+    }
+    return elementAttributes;
+};
+
+export const viewModelFunction = (model: Button) => {
+    let icon;
+    if(model.icon || model.type === 'back') {
+        icon = getImageContainerJSX(model.icon || 'back');
+    }
+    const elementAttributes = getAttributes(model.elementAttr);
+
+    return {
+        ...model,
+        elementAttributes,
+        cssClasses: getCssClasses(model),
+        style: {
+            width: model.width,
+            height: model.height
+        },
+        icon,
+    };
+}
+
+export const viewFunction = (viewModel: Button & { cssClasses: string, style: { width?: string } }) => (
+    <div
+        {...viewModel.elementAttr}
+        className={viewModel.cssClasses}
+        title={viewModel.hint}
+        style={viewModel.style}
+        hidden={!viewModel.visible}
+        onPointerOver={viewModel.onPointerOver}
+        onPointerOut={viewModel.onPointerOut}
+        onPointerDown={viewModel.onPointerDown}
+        onClick={viewModel.onClickHandler}
+    >
+        {viewModel.contentRender && (
+            <div className="dx-button-content">
+                <viewModel.contentRender icon={viewModel.icon} text={viewModel.text} />
+            </div>
+        ) || (
+            <div className="dx-button-content">
+                {viewModel.icon}
+                {viewModel.text && <span className="dx-button-text">{viewModel.text}</span>}
+            </div>
+        )}
+
+    </div>
+);
+
+@Component({
+    name: 'Button',
+    components: [],
+    viewModel: viewModelFunction,
+    view: viewFunction,
+})
+
+export default class Button {
+    @Prop() classNames?: string[]
+    @Prop() height?: string;
+    @Prop() hint?: string;
+    @Prop() icon?: string;
+    @Prop() pressed?: boolean;
+    @Prop() stylingMode?: string;
+    @Prop() text?: string;
+    @Prop() type?: string;
+    @Prop() width?: string;
+    @Prop() elementAttr?: { [name: string]: any } = {};
+    @Prop() visible?: boolean = true;
+
+    // @Template() contentRender?: any;
```

Comment:
Forgotten comment?

Reply:
Should be Template, but the generator doesn't support it at the moment. I left the comment to no forget

Reply:
removed